pkg/client/version_test.go

```diff
@@ -0,0 +1,136 @@
+package client_test
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/telepresenceio/telepresence/v2/pkg/client"
+)
+
+func TestGetInstallMechanism(t *testing.T) {
+	type testcase struct {
+		binaryPath               string
+		symLinkPath              string
+		expectedInstallMechanism string
+		errFile                  bool
+	}
+	fakeExecDir := t.TempDir()
+	testcases := map[string]testcase{
+		"website-install": {
+			binaryPath:               "telepresence",
+			symLinkPath:              "",
+			expectedInstallMechanism: "website",
+			errFile:                  false,
+		},
+		"docker-install": {
+			binaryPath:               "docker/telepresence",
+			symLinkPath:              "",
+			expectedInstallMechanism: "docker",
+			errFile:                  false,
+		},
+		"docker-install-sym": {
+			binaryPath:               "docker/telepresence",
+			symLinkPath:              "telepresence",
+			expectedInstallMechanism: "docker",
+			errFile:                  false,
+		},
+		// we care about the underlying executable so even if someone
+		// symlinks to make it *seem* like it's installed via docker
+		// it will use the actual executable.
+		"pseudo-symlink": {
+			binaryPath:               "telepresence",
+			symLinkPath:              "docker/telepresence",
+			expectedInstallMechanism: "website",
+			errFile:                  false,
+		},
+		"fail-executable-files": {
+			binaryPath:               "telepresence",
+			symLinkPath:              "",
+			expectedInstallMechanism: "undetermined",
+			errFile:                  true,
+		},
+	}
+	if runtime.GOOS == "darwin" {
+		testcases["brew-install"] = testcase{
+			binaryPath:               "Cellar/telepresence",
+			symLinkPath:              "",
+			expectedInstallMechanism: "brew",
+			errFile:                  false,
+		}
+		testcases["brew-install-sym"] = testcase{
+			binaryPath:               "Cellar/telepresence",
+			symLinkPath:              "telepresence",
+			expectedInstallMechanism: "brew",
+			errFile:                  false,
+		}
+	} else {
+		// I'm not sure why anybody would have brew paths in a non-macos
+		// but brew is *only* for macos so we should report website in
+		// this case.
```

Comment:
well there is brew for linux, but i suppose we don't support it, given that our brew formulae just download the binary

Reply:
TIL, I will update that comment so it's accurate 😬